react/core/main.tsx

```diff
@@ -63,6 +65,14 @@ import {
 import withHMR from '../utils/withHMR'
 import { generateExtensions } from '../utils/blocks'
 import { promised } from '../utils/promise'
+import {
+  RenderedSuccess,
+  NamedServerRendered,
+  Rendered,
+  ServerRendered,
+} from '../typings/global'
+import { RenderRuntime, Extensions } from '../typings/runtime'
+import '../typings/runtime'
```

Comment:
nit: what do you think about starting to use the `type` modifier? `import type { ... } from ...`?